mula/tests/integration/test_boefje_scheduler.py

```diff
@@ -830,6 +911,203 @@ def test_post_push(self):
+            hash=boefje_task1.hash,
+            data=boefje_task1.model_dump(),
+            created_at=datetime.now(timezone.utc),
+            modified_at=datetime.now(timezone.utc),
+        )
+
+        task2 = models.Task(
+            scheduler_id=self.scheduler.scheduler_id,
+            priority=1,
+            type=models.BoefjeTask.type,
+            hash=boefje_task2.hash,
+            data=boefje_task2.model_dump(),
+            created_at=datetime.now(timezone.utc),
+            modified_at=datetime.now(timezone.utc),
+        )
+
+        item1 = functions.create_item(
+            scheduler_id=self.organisation.id,
+            priority=1,
+            task=task1,
+        )
+
+        item2 = functions.create_item(
+            scheduler_id=self.organisation.id,
+            priority=1,
+            task=task2,
+        )
+
+        self.mock_get_plugin.return_value = PluginFactory(
+            scan_level=0,
+            consumes=[ooi1.object_type],
+            cron=cron,
+        )
+
+        # Act
+        self.scheduler.push_item_to_queue(item1)
+        self.scheduler.push_item_to_queue(item2)
+
+        # Tasks should be on priority queue
+        task_pq1 = models.BoefjeTask(**self.scheduler.queue.peek(0).data)
+        task_pq2 = models.BoefjeTask(**self.scheduler.queue.peek(1).data)
+        self.assertEqual(2, self.scheduler.queue.qsize())
+        self.assertEqual(ooi1.primary_key, task_pq1.input_ooi)
+        self.assertEqual(boefje_task1.boefje.id, task_pq1.boefje.id)
+        self.assertEqual(ooi2.primary_key, task_pq2.input_ooi)
+        self.assertEqual(boefje_task2.boefje.id, task_pq2.boefje.id)
+
+        # Task should be in datastore, and queued
+        task_db1 = self.mock_ctx.datastores.task_store.get_task(item1.id)
+        task_db2 = self.mock_ctx.datastores.task_store.get_task(item2.id)
+        self.assertEqual(task_db1.id, item1.id)
+        self.assertEqual(task_db1.status, models.TaskStatus.QUEUED)
+        self.assertEqual(task_db2.id, item2.id)
+        self.assertEqual(task_db2.status, models.TaskStatus.QUEUED)
+
+        # Schedule should be in datastore
+        schedule_db1 = self.mock_ctx.datastores.schedule_store.get_schedule(task_db1.schedule_id)
+        schedule_db2 = self.mock_ctx.datastores.schedule_store.get_schedule(task_db2.schedule_id)
+        self.assertIsNotNone(schedule_db1)
+        self.assertEqual(schedule_db1.id, task_db1.schedule_id)
+        self.assertIsNotNone(schedule_db2)
+        self.assertEqual(schedule_db2.id, task_db2.schedule_id)
+
+        # Schedule deadline should be set
+        self.assertIsNotNone(schedule_db1.deadline_at)
+        self.assertIsNotNone(schedule_db2.deadline_at)
+
+        # Schedule cron should be set
+        self.assertIsNotNone(schedule_db1.schedule)
+        self.assertEqual(schedule_db1.schedule, cron)
+        self.assertIsNotNone(schedule_db2.schedule)
+        self.assertEqual(schedule_db2.schedule, cron)
```

Comment:
Perhaps there is some redundancy here given your other tests, although I haven't read these very carefully. It's tricky but my preference is always to keep the tests as orthogonal as possible as well :slightly_smiling_face:

Reply:
Yeah I agree, I think I'll delete this test. It doesn't add more than the test with a single item.